src/services/qix/session.js

```diff
@@ -133,6 +170,143 @@ class Session {
   }
 
   /**
+  * Function used to suspend the session.
+  * @returns {Object} Returns a promise instance.
+  */
+  suspend() {
+    return this.rpc.close(MANUAL_SUSPEND);
+  }
+
+  /**
+  * Function used to restore the global API.
+  * @param {Object} restoredApis - The API-cache containing restored APIs.
+  * @returns {Object} Returns a promise instance.
+  */
+  restoreGlobal(restoredApis) {
+    const global = this.apis.getApisByType('Global').pop();
+    restoredApis.add(global.handle, global.api);
+    return this.Promise.resolve();
+  }
+
+  /**
+  * Function used to restore the doc API.
+  * @param {String} sessionState - The state of the session, attached or created.
+  * @param {Object} restoredApis - The API-cache containing restored APIs.
+  * @returns {Object} Returns a promise instance.
+  */
+  restoreDoc(sessionState, restoredApis) {
+    const app = this.apis.getApisByType('Doc').pop();
```

Comment:
We could remove the `sessionState` parameter here and always try to restore the active doc or open if it fails. Simplifies the flow in `resume()`.

```js
  restoreDoc(sessionState, restoredApis) {
    const app = this.apis.getApisByType('Doc').pop();

    if (!app) {
      return this.Promise.resolve();
    }

    return this.rpc.send({
      method: 'GetActiveDoc',
      handle: -1,
      params: [],
    }).then((response) => {
      if (response.error) {
        return this.rpc.send({
          method: 'OpenDoc',
          handle: -1,
          params: [this.appId, '', '', '', !!this.noData],
        });
      }
      return response;
    }).then((response) => {
      if (response.error) {
        return this.Promise.reject(new Error(response.error.message));
      }
      const handle = response.result.qReturn.qHandle;
      app.api.handle = handle;
      restoredApis.add(handle, app.api);
      return app.api;
    });
  }
```